src/XMakeTasks/Copy.cs

```diff
@@ -323,20 +332,52 @@ private void LogDiagnostic(string message, params object[] messageArgs)
                     FileUtilities.DeleteNoThrow(destinationFileState.Name);
                 }
 
-                hardLinkCreated = NativeMethods.CreateHardLink(destinationFileState.Name, sourceFileState.Name, IntPtr.Zero /* reserved, must be NULL */);
+                linkCreated = NativeMethods.CreateHardLink(destinationFileState.Name, sourceFileState.Name, IntPtr.Zero /* reserved, must be NULL */);
 
-                if (!hardLinkCreated)
+                if (!linkCreated)
                 {
                     int errorCode = Marshal.GetHRForLastWin32Error();
                     Exception hardLinkException = Marshal.GetExceptionForHR(errorCode);
                     // This is only a message since we don't want warnings when copying to network shares etc.
                     Log.LogMessageFromResources(MessageImportance.Low, "Copy.RetryingAsFileCopy", sourceFileState.Name, destinationFileState.Name, hardLinkException.Message);
                 }
             }
+            else if (UseSymboliclinksIfPossible)
```

Comment:
Hi @cdmihai. Something like this?
```c#
 // If we want to create hard or symbolic links, then try that first
            if (UseHardlinksIfPossible || UseSymboliclinksIfPossible)
            {
                // Do not log a fake command line as well, as it's superfluous, and also potentially expensive
                Log.LogMessageFromResources(MessageImportance.Normal, true == UseHardlinksIfPossible ? "Copy.HardLinkComment" : "Copy.SymbolicLinkComment", sourceFileState.Name, destinationFileState.Name);

                if (!_overwriteReadOnlyFiles)
                {
                    destinationFileExists = destinationFileState.FileExists;
                }

                // CreateHardLink and CreateSymbolicLink cannot overwrite an existing file or link
                // so we need to delete the existing entry before we create the hard or symbolic link.
                // We need to do a best-effort check to see if the files are the same
                // if they are the same then we won't delete, just in case they refer to the same
                // physical file on disk.
                // Since we'll fall back to a copy (below) this will fail and issue a correct
                // message in the case that the source and destination are in fact the same file.
                if (destinationFileExists && !IsMatchingSizeAndTimeStamp(sourceFileState, destinationFileState))
                {
                    FileUtilities.DeleteNoThrow(destinationFileState.Name);
                }

                linkCreated = true == UseHardlinksIfPossible? NativeMethods.CreateHardLink(destinationFileState.Name, sourceFileState.Name, IntPtr.Zero /* reserved, must be NULL */) : NativeMethods.CreateSymbolicLink(destinationFileState.Name, sourceFileState.Name, SymbolicLink.File);

                if (!linkCreated)
                {
                    int errorCode = Marshal.GetHRForLastWin32Error();
                    Exception linkException = Marshal.GetExceptionForHR(errorCode);
                    // This is only a message since we don't want warnings when copying to network shares etc.
                    Log.LogMessageFromResources(true == UseHardlinksIfPossible ? MessageImportance.Low : MessageImportance.High, "Copy.RetryingAsFileCopy", sourceFileState.Name, destinationFileState.Name, linkException.Message);
                }
            }
```